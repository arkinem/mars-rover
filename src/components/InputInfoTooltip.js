import React from "react";
import styled from "styled-components";
import { FaInfoCircle } from "react-icons/fa";
import { colors, shadow } from "../helpers/style";

export default () => (
  <Container>
    <InfoIcon />
    <TooltipContainer>
      <Paragraph>
        The <B>first line of input</B> is the upper-right coordinates of the
        plateau, the lower-left coordinates are assumed to be 0,0.
      </Paragraph>
      <Paragraph>
        The rest of the input is information pertaining to the rovers that have
        been deployed. Each rover has two lines of input. The first line gives
        the rover's position, and the second line is a series of instructions
        telling the rover how to explore the plateau.
      </Paragraph>
      <Paragraph>
        The position is made up of two integers and a letter separated by
        spaces, corresponding to the x and y co-ordinates and the rover's
        orientation.
      </Paragraph>
      <Paragraph>
        Example:{"\n"}5 5{"\n"}2 N LMLMLMLMM{"\n"}3 3 E MMRMMRMRRM
      </Paragraph>
      <Paragraph>
        If you need more help, look here:{"\n"}
        <Link
          target="_blank"
          href="https://code.google.com/archive/p/marsrovertechchallenge/"
        >
          https://code.google.com/archive/p/marsrovertechchallenge/
        </Link>
      </Paragraph>
    </TooltipContainer>
  </Container>
);

const Container = styled.div`
  position: absolute;
  top: 5px;
  right: 5px;
  height: 30px;
  width: 30px;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: all 0.3s;
`;

const InfoIcon = styled(FaInfoCircle)`
  ${Container}:hover & {
    opacity: 0.8;
  }
`;

const TooltipContainer = styled.div`
  position: absolute;
  max-width: 270px;
  width: 270px;
  right: 30px;
  top: -40px;
  padding: 12px;
  z-index: 90;
  background: ${colors.surface};
  border-radius: 4px;
  box-shadow: ${shadow.l};

  display: none;
  ${Container}:hover & {
    display: flex;
    flex-direction: column;
  }
`;

const Paragraph = styled.p`
  font-size: 14px;
  line-height: 18px;
  font-weight: 300;
  margin-bottom: 12px;
  white-space: pre-wrap;
`;

const B = styled.span`
  font-size: 500;
`;

const Link = styled.a`
  overflow-wrap: break-word;
  color: ${colors.text};
  text-decoration: none;

  opacity: 0.9;
  :hover {
    opacity: 0.5;
  }
`;
