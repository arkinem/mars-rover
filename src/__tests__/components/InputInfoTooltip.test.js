import React from "react";
import renderer from "react-test-renderer";
import InputInfoTooltip from "../../components/InputTooltip";

describe("InputInfoTooltip", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<InputInfoTooltip />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
