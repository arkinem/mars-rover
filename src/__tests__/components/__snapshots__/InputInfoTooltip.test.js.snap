// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputInfoTooltip renders correctly 1`] = `
<div
  className="sc-AxjAm hXgkib"
>
  <svg
    className="sc-AxirZ jzxuBH"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
    />
  </svg>
  <div
    className="sc-AxiKw jxkrOj"
  >
    <p
      className="sc-AxhCb ixBvKy"
    >
      The 
      <span
        className="sc-AxhUy bHFpOQ"
      >
        first line of input
      </span>
       is the upper-right coordinates of the plateau, the lower-left coordinates are assumed to be 0,0.
    </p>
    <p
      className="sc-AxhCb ixBvKy"
    >
      The rest of the input is information pertaining to the rovers that have been deployed. Each rover has two lines of input. The first line gives the rover's position, and the second line is a series of instructions telling the rover how to explore the plateau.
    </p>
    <p
      className="sc-AxhCb ixBvKy"
    >
      The position is made up of two integers and a letter separated by spaces, corresponding to the x and y co-ordinates and the rover's orientation.
    </p>
    <p
      className="sc-AxhCb ixBvKy"
    >
      Example:
      

      5 5
      

      2 N LMLMLMLMM
      

      3 3 E MMRMMRMRRM
    </p>
    <p
      className="sc-AxhCb ixBvKy"
    >
      If you need more help, look here:
      

      <a
        className="sc-AxgMl gWEYHI"
        href="https://code.google.com/archive/p/marsrovertechchallenge/"
        target="_blank"
      >
        https://code.google.com/archive/p/marsrovertechchallenge/
      </a>
    </p>
  </div>
</div>
`;
